5calls/app/src/main/java/org/a5calls/android/a5calls/controller/LocationActivity.java

```diff
@@ -137,20 +136,9 @@ private void submitZip() {
     }
 
     private void returnToMain() {
-        // Make sure we're still alive
-        if (isFinishing() || isDestroyed()) {
-            return;
-        }
-
-        // If we came from MainActivity and return with another Intent, it will create a deep stack
-        // of activities!
-        if (mFromMain) {
-            onBackPressed();
-        } else {
-            Intent intent = new Intent(this, MainActivity.class);
-            startActivity(intent);
-            finish();
-        }
+        Intent intent = new Intent(this, MainActivity.class);
```

Comment:
So I think that if you return always with an intent, we end up with a big stack of activities (Location / Main / Location / Main). Even if you finish(), you end up with a stack of Main activities.

Curious why you changed this from the if (mFromMain) { onBackPressed() }?